### PR TITLE
Add DDNS API cdmon.org

### PR DESCRIPTION
**Pull request by Serrallonga25**
Added support for DDNS API https://dinamico.cdmon.org

### CODE REVIEW

File: ddclient.in [hunk 1]
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
...
+  protocol=cdmon               ##
+  login=service-login          ## login name and password registered with the service
+  password=service-password    ##
+  fully.qualified.host         ## the host registered with the service.
+
+Example ${program}.conf file entries:
+  ## single host update
+  protocol=cdmon,                                               \\
+  login=my-my-cdmon.org-login,                                     \\
+  password=my-cdmon.org-password                                   \\
```
Comment (SuperSandro2000) [inline ddclient.in:5483]:
```suggestion
  protocol=cdmon,                                               \\
  login=my-my-cdmon.org-login,                                  \\
  password=my-cdmon.org-password                                \\
```

File: ddclient.in [hunk 2]
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
...
+  myhost.mydomain.cat
```
Comment (SuperSandro2000) [inline ddclient.in:5484]:
```suggestion
  domain.example.com
```

File: ddclient.in [hunk 3]
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
...
+##
+## adapted by David
+## https://ticket.cdmon.com/ca/support/solutions/articles/7000005922-api-d-actualitzaci%c3%b3-de-ip-del-dns-gratis-din%c3%a0mic?set_locale=1075
+## https://dinamico.cdmon.org/onlineService.php?enctype=MD5&n=usuario&p=1bc29b36f623ba82aaf6724fd3b16718&cip=x.x.x.x
+##
+######################################################################
+sub dinamico_cdmon_update {
+
+
+    debug("\ndinamico_cdmon_update -------------------");
```
Comment (SuperSandro2000) [inline ddclient.in:5499]:
```suggestion
    debug("\ndinamico_cdmon_update -------------------");
```

File: ddclient.in [hunk 4]
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
...
+
+    ## update each configured host
+    foreach my $h (@_) {
+        my $ip = delete $config{$h}{'wantip'};
+        info("setting IP address to %s for %s", $ip, $h);
+        verbose("UPDATE:","updating %s", $h);
```
Comment (SuperSandro2000) [inline ddclient.in:5505]: This is duplicated

File: ddclient.in [hunk 5]
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
...
+
+        my $url;
+        $url   = "https://dinamico.cdmon.org/onlineService.php";
+        $url  .= "?enctype=MD5";
+        $url  .= "&n=$config{$h}{'login'}";
+        $url  .= "&p=$config{$h}{'password'}";
+        $url  .= "&cip=$ip" if $ip;
```
Comment (SuperSandro2000) [inline ddclient.in:5512]:
```suggestion
        my $url = "https://dinamico.cdmon.org/onlineService.php?enctype=MD5&n=$config{$h}{'login'}&p=$config{$h}{'password'}";
        $url  .= "&cip=$ip" if $ip;
```

Can we use something other than MD5? MD5 is insecure for cryptography.

File: ddclient.in [hunk 6]
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
...
+                if (exists $params->{'customip'}) {
+                    $msg  .= sprintf " : sent IP address %s", $params->{'customip'};
+                } else {
+                    $msg  .= sprintf " : sent IP address %s", $ip;
+                }
+
+                if (exists $params->{'newip'})    {$newip = $params->{'newip'}; $msg  .= sprintf " :IP address set to newip:%s", $newip;}
+                if (exists $params->{'temps'})    {$msg  .= sprintf ", the changes will not be operational until after %d minutes", int(int($params->{'temps'})/60000);}
+
+                #$config{$h}{'ip'}     = $ip;
```
Comment (SuperSandro2000) [inline ddclient.in:5567]:
```suggestion
```

File: ddclient.in [hunk 7]
```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
...
+                $msg  = sprintf "updating %s: good", $h;
+
+                if (exists $params->{'customip'}) {
+                    $msg  .= sprintf " : sent IP address %s", $params->{'customip'};
+                } else {
+                    $msg  .= sprintf " : sent IP address %s", $ip;
+                }
+
+                if (exists $params->{'newip'})    {$newip = $params->{'newip'}; $msg  .= sprintf " :IP address set to newip:%s", $newip;}
+                if (exists $params->{'temps'})    {$msg  .= sprintf ", the changes will not be operational until after %d minutes", int(int($params->{'temps'})/60000);}
```
Comment (SuperSandro2000) [inline ddclient.in:5565]:
```suggestion
                if (exists $params->{'newip'}) {
                  $newip = $params->{'newip'};
                  $msg  .= sprintf " :IP address set to newip:%s", $newip;
                }
                if (exists $params->{'temps'}) {
                  $msg  .= sprintf ", the changes will not be operational until after %d minutes", int(int($params->{'temps'})/60000);
                }
```

File: ddclient.in (hunk 5 above)
Comment (Serrallonga25) [reply ddclient.in:5510]: It is the only thing that the provider accepts, the password must be in MD5

File: ddclient.in (hunk 4 above)
Comment (Serrallonga25) [reply ddclient.in:5503]: I left it that way because I saw that both “sub nic_changeip_update {” line 5656 and “sub nic_1984_update {” line 5426 have it

**Comment by Serrallonga25** [comment]
Hello I have added and modified some changes in the descriptive text
When downloading and compiling the code I get an error that did not come out before.

WARNING:  skipping host: my.domain.cat: 'server=1' is an invalid fully qualified host name and optional port number.

**Comment by SuperSandro2000** [comment]
There is another merge conflict. I cannot help you with the error. Please let me know when you fixed it.

**Comment by Serrallonga25** [comment]
> There is another merge conflict. I cannot help you with the error. Please let me know when you fixed it.
Hello SuperSandro, the conflict and the error have already been resolved

**Pull request closed without merging**